core/base/src/TSystem.cxx

```diff
@@ -3770,7 +3798,9 @@ int TSystem::CompileMacro(const char *filename, Option_t *opt,
           gInterpreter->LoadLibraryMap(libmapfilename);
       }
       if (verboseLevel>3 && withInfo)  ::Info("ACLiC","loading the shared library");
-      if (loadLib) result = !gSystem->Load(library);
+      if (loadLib) {
+         return LoadLibrary(library);
```

Comment:
That early return is a change in behavior in the case `if ( !result )` below (line 3806). Is this intended, and why?

Reply:
I think that's a mistake (we will see the tests), good catch.